# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1346202441 14400
#      Tue Aug 28 21:07:21 2012 -0400
# Node ID e787ba62aacf5e7c16eda2245b0eea0cb4cb54d4
# Parent  fa8f3955e70a759d075c3f4ff64f78274c744147
specify separate build dirs for texi2dvi and texi2pdf commands

* acinclude.m4 (OCTAVE_PROG_TEXI2DVI): Append --build-dir option to
TEXI2DVI.
(OCTAVE_PROG_TEXI2PDF): Append --build-dir option to TEXI2PDF.
* doc/interpreter/Makefile.am: Delete commented .NOTPARALLEL
declaration.

diff --git a/doc/interpreter/Makefile.am b/doc/interpreter/Makefile.am
--- a/doc/interpreter/Makefile.am
+++ b/doc/interpreter/Makefile.am
@@ -15,20 +15,16 @@
 # for more details.
 # 
 # You should have received a copy of the GNU General Public License
 # along with Octave; see the file COPYING.  If not, see
 # <http://www.gnu.org/licenses/>.
 
 include $(top_srcdir)/build-aux/common.mk
 
-## Automake generated rules for documentation are not parallel-safe.
-## Restrict current directory to run serially
-## .NOTPARALLEL:
-
 TEXINFO_TEX = ../texinfo.tex
 
 ## Leading PATH_SEPARATOR required due to weak parsing by dvips (12/04/09)
 TEXINPUTS := "$(PATH_SEPARATOR)$(srcdir)$(PATH_SEPARATOR)$(TEXINPUTS)$(PATH_SEPARATOR)"
 export TEXINPUTS
 
 ## Include custom texmf.cnf necessary to run @seealso macro 
 TEXMFCNF := "..$(PATH_SEPARATOR)$(srcdir)/..$(PATH_SEPARATOR)$(TEXMFCNF)$(PATH_SEPARATOR)"
diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -1501,16 +1501,17 @@ AC_DEFUN([OCTAVE_PROG_TEXI2DVI], [
     TEXI2DVI='$(top_srcdir)/build-aux/missing texi2dvi'
     warn_texi2dvi="
 
 I didn't find texi2dvi, but it's only a problem if you need to
 reconstruct the DVI version of the manual
 "
     OCTAVE_CONFIGURE_WARNING([warn_texi2dvi])
   fi
+  TEXI2DVI="$TEXI2DVI"' --build-dir=texi2dvi-build.$$$$'
   AC_SUBST(TEXI2DVI)
 ])
 dnl
 dnl Check for texi2pdf.
 dnl
 AC_DEFUN([OCTAVE_PROG_TEXI2PDF], [
   AC_REQUIRE([OCTAVE_PROG_TEXI2DVI])
   AC_CHECK_PROG(TEXI2PDF, texi2pdf, texi2pdf, [])
@@ -1527,16 +1528,17 @@ AC_DEFUN([OCTAVE_PROG_TEXI2PDF], [
     TEXI2PDF='$(top_srcdir)/build-aux/missing texi2pdf'
     warn_texi2pdf="
 
 I didn't find texi2pdf, but it's only a problem if you need to
 reconstruct the PDF version of the manual
 "
     OCTAVE_CONFIGURE_WARNING([warn_texi2pdf])
   fi
+  TEXI2PDF="$TEXI2PDF"' --build-dir=texi2pdf-build.$$$$'
   AC_SUBST(TEXI2PDF)
 ])
 dnl
 dnl Set default value for a variable and substitute it.
 dnl
 AC_DEFUN([OCTAVE_SET_DEFAULT], [
   ifelse($#, 2, [: ${$1=$2}
 ])dnl
